src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java

```diff
@@ -545,69 +545,47 @@ private void populateSummaryStatisticsFromResponses(
 
             // Compute minimum score received
-            if (!min.containsKey(recipientEmail)) {
-                min.put(recipientEmail, answer);
-            }
-            double minScoreReceived = Math.min(answer, min.get(recipientEmail));
+            double minScoreReceived = Math.min(answer, min.getOrDefault(recipientEmail, answer));
             min.put(recipientEmail, minScoreReceived);
 
             // Compute maximum score received
-            if (!max.containsKey(recipientEmail)) {
-                max.put(recipientEmail, answer);
-            }
-            double maxScoreReceived = Math.max(answer, max.get(recipientEmail));
+            double maxScoreReceived = Math.max(answer, max.getOrDefault(recipientEmail, answer));
             max.put(recipientEmail, maxScoreReceived);
 
             // Compute total score received
-            if (!total.containsKey(recipientEmail)) {
-                total.put(recipientEmail, 0.0);
-            }
-            double totalScore = total.get(recipientEmail) + answer;
+            double totalScore = total.getOrDefault(recipientEmail, 0.0) + answer;
             total.put(recipientEmail, totalScore);
 
             // Compute total score received excluding self
-            if (!totalExcludingSelf.containsKey(recipientEmail)) {
-                totalExcludingSelf.put(recipientEmail, null);
-            }
             if (!isSelfResponse) {
-                Double totalScoreExcludingSelf = totalExcludingSelf.get(recipientEmail);
+                Double totalScoreExcludingSelf = totalExcludingSelf.getOrDefault(recipientEmail, null);
 
                 // totalScoreExcludingSelf == null when the user has only self response
                 totalExcludingSelf.put(recipientEmail,
                                        totalScoreExcludingSelf == null ? answer : totalScoreExcludingSelf + answer);
             }
 
             // Compute average score received
-            if (!average.containsKey(recipientEmail)) {
-                average.put(recipientEmail, 0.0);
-            }
-            double averageReceived = total.get(recipientEmail) / numResponses.get(recipientEmail);
+            double averageReceived = total.getOrDefault(recipientEmail, 0.0) / numResponses.get(recipientEmail);
             average.put(recipientEmail, averageReceived);
 
             // Compute average score received excluding self
-            if (!averageExcludingSelf.containsKey(recipientEmail)) {
-                averageExcludingSelf.put(recipientEmail, null);
-            }
             if (!isSelfResponse && totalExcludingSelf.get(recipientEmail) != null) {
                 double averageReceivedExcludingSelf =
-                        totalExcludingSelf.get(recipientEmail) / numResponsesExcludingSelf.get(recipientEmail);
+                        totalExcludingSelf.get(recipientEmail)
+                        /
+                        numResponsesExcludingSelf.getOrDefault(recipientEmail, null);
```

Comment:
Does not look right here. It is `averageExcludingSelf`, not `numResponsesExcludingSelf`. And again, what happens if `isSelfResponse` is `true`?

Reply:
My mistake! Reverted.